stopes/pipelines/asr_bleu/utils.py

```diff
@@ -0,0 +1,213 @@
+import json
+import re
+import logging
+import urllib.request
+from pathlib import Path
+# import asyncio
+
+import fairseq
+import torch
+# from fairseq.data.data_utils import lengths_to_padding_mask
+from tqdm import tqdm
+
+try:
+    import torchaudio
+    from torchaudio.models.decoder import ctc_decoder
+except ImportError:
+    raise ImportError("Upgrade torchaudio to 0.12 to enable CTC decoding")
+
+logger = logging.getLogger("stopes.asr_bleu.utils")
+
+
+class DownloadProgressBar(tqdm):
+    """A class to represent a download progress bar"""
+
+    def update_to(self, b=1, bsize=1, tsize=None) -> None:
+        """
+        Update the download progress
+        """
+        if tsize is not None:
+            self.total = tsize
+        self.update(b * bsize - self.n)
+
+
+def retrieve_asr_config(lang_key: str, asr_version: str, json_path: str) -> dict:
+    if len(lang_key) !=3:
+        raise ValueError(f"'{lang_key}' lang key for language type must be 3 characters!")
+
+    with open(json_path, "r") as f:
+        asr_model_cfgs = json.load(f)
+    return asr_model_cfgs[lang_key][asr_version]
+
+
+class ASRGenerator(object):
```

Comment:
If the class represents a generator, shouldn't it contain the actual method for generation (i.e. transcription)?

Reply:
Good point. We moved the generation code to transcribe_audio, but the name remained unchanged. Would ASRContainer be a more suitable name? https://github.com/facebookresearch/stopes/pull/52/commits/5e9187eb43e0a29d044ac4e49beb8a41ccbce3a8